File: back/database/migration_bbs.go
package main

import (
	"encoding/json"
	"time"
	"model"
	"io/ioutil"
)

func main() {
	seed()
}

func seed() {
	type Input struct {
		Title       string `json:"title"`
		Contributor string `json:"name"`
		Content     string `json:"content"`
		Date        string `json:"date"`
		Password    string `json:"password"`
	}

	var input []Input
	file, _ := ioutil.ReadFile("database/data/bbs-data.json")
	json.Unmarshal(file, &input)

	db := model.Connect()
	for _, v := range input {
		tmpDate, _ := time.Parse("2006/01/02 15:04:05", v.Date)
		db.Create(&model.Bbs{
			Title      : v.Title,
			Contributor: v.Contributor,
			Content    : v.Content,
			Password   : v.Password,
			IsDelete   : false,
			CreatedAt  : tmpDate,
			UpdatedAt  : tmpDate,
		})
	}
}

File: front/database/migration_bbs.js
const fs = require('fs')
const firebase = require('firebase')
const config = require('../slf-config')

require("firebase/firestore")
firebase.initializeApp(config.firebase)
const db = firebase.firestore()
const settings = {timestampsInSnapshots: true};
db.settings(settings);

const filePath = __dirname + '/bbs-data.json'
const json = JSON.parse(fs.readFileSync(filePath, 'utf-8'))

json.forEach((v) => {
  const tmp = new Date(v.date)
  const key = String(tmp.getTime())
  db.collection('bbs').doc(key).set({
    title: v.title,
    author: v.name,
    content: v.content,
    password: parseInt(v.password),
    createdAt: tmp,
    updatedAt: tmp,
  })
})